Shoko.Server/API/v2/Modules/PlexWebhook.cs

```diff
@@ -1,348 +1,643 @@
+                    TraktScrobble(payload, ScrobblePlayingStatus.Pause);
+                    break;
+                case "media.stop":
+                    TraktScrobble(payload, ScrobblePlayingStatus.Stop);
+                    break;
+            }
+
+            return Ok(); //doesn't need to be an ApiStatus.OK() since really, all I take is plex.
+        }
+
+        #region Plex events
+
+        [NonAction]
+        private void TraktScrobble(PlexEvent evt, ScrobblePlayingStatus type)
+        {
+            var metadata = evt.Metadata;
+            var (episode, anime) = GetEpisodeAndSeries(metadata);
+
+            if (episode == null) return;
+
+            var vl = RepoFactory.VideoLocal.GetByAniDBEpisodeID(episode.AniDB_EpisodeID).FirstOrDefault();
+
+            float per = 100 * (metadata.ViewOffset / (float)vl.Duration); //this will be nice if plex would ever give me the duration, so I don't have to guess it.
+
+            ScrobblePlayingType scrobbleType = episode.GetAnimeSeries()?.GetAnime()?.AnimeType == (int) AnimeType.Movie
+                ? ScrobblePlayingType.movie
+                : ScrobblePlayingType.episode;
+
+            TraktTVHelper.Scrobble(scrobbleType, episode.AnimeEpisodeID.ToString(), type, per);
+        }
+
+        [NonAction]
+        private void Scrobble(PlexEvent data)
+        {
+            PlexEvent.PlexMetadata metadata = data.Metadata;
+            (SVR_AnimeEpisode episode, SVR_AnimeSeries anime) = GetEpisodeAndSeries(metadata);
+
+            if (episode == null)
+            {
+                Logger.LogInformation("No episode returned, aborting scrobble. This might not have been a ShokoMetadata library");
+                return;
+            }
+
+            Logger.LogTrace("Got anime: {AnimeName}, episode: {EpisodeNumber}", anime, episode.AniDB_Episode.EpisodeNumber);
+            var user = RepoFactory.JMMUser.GetAll().FirstOrDefault(u => data.Account.Title.FindIn(u.GetPlexUsers()));
+            if (user == null)
+            {
+                Logger.LogInformation("Unable to determine who {AccountName} is in Shoko, make sure this is set under user settings in Desktop", data.Account.Title);
+                return; //At this point in time, we don't want to scrobble for unknown users
+            }
+
+            episode.ToggleWatchedStatus(true, true, FromUnixTime(metadata.LastViewedAt), false, user.JMMUserID,
+                true);
+            anime.UpdateStats(true, false, true);
+        }
+
+        #endregion
+
+        [NonAction]
+        private (SVR_AnimeEpisode, SVR_AnimeSeries) GetEpisodeAndSeries(PlexEvent.PlexMetadata metadata)
+        {
+            Uri guid = new Uri(metadata.Guid);
+            if (guid.Scheme != "com.plexapp.agents.shoko") return (null, null);
+
+            PathString ps = guid.AbsolutePath;
+
+            int animeId = int.Parse(guid.Authority);
+            int series = int.Parse(guid.AbsolutePath.Split('/')[1]);

```

Comment:
This is the season number, not the series. Should we maybe change this variable name?

Reply:
Sure, we can do that.
I tried to refrain from touching more of the code then needed, but this change will make it easier for a future maintainer to read the code instead of getting confused.